src/commonMain/kotlin/exchange.dydx.abacus/state/v2/supervisor/VaultSupervisor.kt

```diff
@@ -132,18 +137,25 @@ internal class VaultSupervisor(
     }
 
     private fun retrieveMegaVaultPnl() {
-        val url = helper.configs.publicApiUrl("vaultHistoricalPnl")
-        if (url != null) {
-            helper.get(
-                url = url,
-                params = mapOf("resolution" to "day"),
-                headers = null,
-            ) { _, response, httpCode, _ ->
-                if (helper.success(httpCode) && response != null) {
-                    stateMachine.onMegaVaultPnl(response)
-                } else {
+        val scope = CoroutineScope(Dispatchers.Unconfined)
+        scope.launch {
+            val url = helper.configs.publicApiUrl("vaultHistoricalPnl")
+            if (url != null) {
+                val deferredDaily = async { helper.getAsync(url, params = mapOf("resolution" to "day"), headers = null) }
+                val deferredHourly = async { helper.getAsync(url, params = mapOf("resolution" to "hour"), headers = null) }
+
+                val dailyResponse = deferredDaily.await()
+                val hourlyResponse = deferredHourly.await()
+
+                if (dailyResponse.response != null && hourlyResponse.response != null) {
+                    stateMachine.onMegaVaultPnl(listOf(dailyResponse.response, hourlyResponse.response))
+                } else if (dailyResponse.error != null) {
```

Comment:
Can we still process the valid response if one of requests fails?  It's better than throwing errors.

Not sure about the latency, but it might be better to request the hourly after the daily so that the UI shows the data faster.

Reply:
seem to have similar latency 
![Screenshot 2024-10-23 at 12 10 28 PM](https://github.com/user-attachments/assets/0f8cb595-b855-4f2e-b161-309f20d3f072)

Reply:
will change to process the single failure differently though